R/mrmash_wrapper.R

```diff
@@ -126,20 +127,23 @@ mrmash_wrapper <- function(X,
   }
   prior_grid <- compute_grid(bhat = sumstats$Bhat, sbhat = sumstats$Shat)
 
+  # Compute canonical matrices, if requested
   # Compute canonical matrices, if requested
   if (isTRUE(prior_canonical_matrices)) {
-    prior_canonical_matrices <- compute_canonical_covs(ncol(Y),
-      singletons = TRUE,
-      hetgrid = c(0, 0.25, 0.5, 0.75, 1)
-    )
-    if (!is.null(prior_data_driven_matrices)) {
-      prior_data_driven_matrices <- filter_datadriven_mats(Y, prior_data_driven_matrices)
-      S0_raw <- c(prior_canonical_matrices, prior_data_driven_matrices)
+    prior_canonical_matrices <- compute_canonical_covs(ncol(Y), singletons = TRUE, 
+                                                       hetgrid = c(0, 0.25, 0.5, 0.75, 1))
+    if (!is.null(prior_data_driven)) {
+      names(prior_data_driven)[1:2] <- c("weights","matrices")
+      prior_data_driven <- create_mixture_prior(mixture_prior = prior_data_driven,weights_tol = weights_tol,include_indices = colnames(Y))
+      print(prior_data_driven[["prior_variance"]][["xUlist"]])
+      S0_raw <- c(prior_canonical_matrices, prior_data_driven[["prior_variance"]][["xUlist"]][-1])
     } else {
       S0_raw <- prior_canonical_matrices
     }
   } else {
-    S0_raw <- filter_datadriven_mats(Y, prior_data_driven_matrices)
+      names(prior_data_driven)[1:2] <- c("weights","matrices")
```

Comment:
same comment as above. We cannot rely on the default order that the 1st is w 2nd is U. It's risky